### PR TITLE
UI: Check VC++ Runtime version on startup

### PR DESCRIPTION
**Pull request by derrod**
### Description

Adds a check on startup that ensures a compatible Visual C++ runtime is being used.

The user is given the option to automatically open our existing VC runtime download page (https://obsproject.com/visual-studio-2019-runtimes),

### Motivation and Context

Due to backwards-incompatible changes in MSVC 14.40 OBS crashes on startup for some users that do not have up to date runtimes installed. See #10799.

We should also adjust the updater and installer to check for these older versions, but that will be a larger undertaking and also does not cover the use case of portable ZIPs (unlike this PR).

### How Has This Been Tested?

Tested on Windows VM with outdated vc runtime:

warning

### Types of changes

- Tweak (non-breaking change to improve existing functionality)

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

File: UI/obs-app.cpp [hunk 1]
```diff
@@ -2901,9 +2902,41 @@ void OBSApp::commitData(QSessionManager &manager)
...
+}
+#endif
+
 int main(int argc, char *argv[])
 {
-#ifndef _WIN32
+#ifdef _WIN32
+	if (is_outdated_msvc())
+		return 1;
+#elif defined(_WIN32)
```
Comment (notr1ch) [inline UI/obs-app.cpp:2939]:
```suggestion
#else
```

Comment (derrod) [reply UI/obs-app.cpp]: Whoops, fixed.

File: UI/obs-app.cpp [hunk 2]
```diff
@@ -2901,6 +2902,35 @@ void OBSApp::commitData(QSessionManager &manager)
 }
 #endif
 
+#ifdef _WIN32
+static constexpr char vcRunErrorTitle[] = "Outdated Visual C++ Run Version";
```
Comment (RytoEX) [inline UI/obs-app.cpp:2906]:
```suggestion
static constexpr char vcRunErrorTitle[] = "Outdated Visual C++ Runtime Version";
```

**Review by RytoEX — CHANGES_REQUESTED** [review]

**Comment by notr1ch** [comment]
Perhaps the wording could be changed to "You will now be directed to the download page." with OK / Cancel buttons. Currently it's unclear that pressing "No" will exit OBS.

**Comment by derrod** [comment]
I updated the wording (not the screenshot though). I also wasn't super happy with it, but I didn't have any better ideas. Perhaps @Warchamp7 has a suggestion as well.

Either way I'd like to have this for beta2 (alongside the updater/installer changes) so we can mitigate the VC runtime mess as much as possible.

**Review by RytoEX — APPROVED** [review]
Maybe there's a better way to handle this, but if we figure that out, we can make further changes later. Let's mitigate this as much as we can.

**Comment by wang-bin** [comment]
I my project I find out the std::mutex is incompatible with old msvcp140.dll, I tested with the simplest mutext code. Adding `-D_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR` can fix the incompatibility

**Comment by RytoEX** [comment]
> I my project I find out the std::mutex is incompatible with old msvcp140.dll, I tested with the simplest mutext code. Adding `-D_DISABLE_CONSTEXPR_MUTEX_CONSTRUCTOR` can fix the incompatibility

We considered this and decided against it.

**Pull request merged**